pkg/resource/table/hooks.go

```diff
@@ -0,0 +1,97 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package table
+
+import (
+	"errors"
+	"time"
+
+	"github.com/aws-controllers-k8s/dynamodb-controller/apis/v1alpha1"
+	ackrequeue "github.com/aws-controllers-k8s/runtime/pkg/requeue"
+)
+
+var (
+	ErrTableDeleting = errors.New("table in 'DELETING' state, cannot be modified or deleted")
+	ErrTableCreating = errors.New("table in 'CREATING' state, cannot be modified or deleted")
+	ErrTableUpdating = errors.New("table in 'UPDATING' state, cannot be modified or deleted")
+)
+
+var (
+	// TerminalStatuses are the status strings that are terminal states for a
+	// DynamoDB table
+	TerminalStatuses = []v1alpha1.TableStatus_SDK{
+		v1alpha1.TableStatus_SDK_ARCHIVING,
+		v1alpha1.TableStatus_SDK_DELETING,
+	}
+)
+
+var (
+	requeueWaitWhileDeleting = ackrequeue.NeededAfter(
+		ErrTableDeleting,
+		5*time.Second,
+	)
+	requeueWaitWhileCreating = ackrequeue.NeededAfter(
+		ErrTableDeleting,
+		5*time.Second,
+	)
+	requeueWaitWhileUpdating = ackrequeue.NeededAfter(
+		ErrTableUpdating,
+		5*time.Second,
+	)
+)
+
+// tableHasTerminalStatus returns whether the supplied Dynamodb table is in a
+// terminal state
+func tableHasTerminalStatus(r *resource) bool {
+	if r.ko.Status.TableStatus == nil {
+		return false
+	}
+	ts := *r.ko.Status.TableStatus
+	for _, s := range TerminalStatuses {
+		if ts == string(s) {
+			return true
+		}
+	}
+	return false
+}
+
+// tableCreating returns true if the supplied DynamodbDB table is in the process
+// of being created
+func tableCreating(r *resource) bool {
```

Comment:
nit: (and for `tableDeleting` and `tableUpdating`) 
```suggestion
func isTableCreating(r *resource) bool {
```

Getters returning booleans to start with `is`